cmd/things-jwt/main.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 ForgeRock AS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"crypto"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	frcrypto "github.com/ForgeRock/iot-edge/v7/internal/crypto"
+	"github.com/ForgeRock/iot-edge/v7/internal/jws"
+	"github.com/jessevdk/go-flags"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+func createThingJWT(key crypto.Signer, url string, version string, n int64, custom map[string]interface{}) (string, error) {
+	opts := &jose.SignerOptions{}
```

Comment:
@JacoJooste 
I am edging towards adding the `"typ":"JWT"` header. According to the [spec](https://datatracker.ietf.org/doc/html/rfc7519#section-5.1) it is optional and we obviously allow it but for best practice we should probably add it. What do you think?

Reply:
If it's optional and we don't process it on AM then I would leave it out. It just increases the size unnecessarily and it's another thing to document and maintain in my opinion.

Reply:
Good point about the size. I will remove it from the authentication JWT for consistency.